shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/core/metadata/model/PipelineTableMetaDataTest.java

```diff
@@ -0,0 +1,50 @@
+package org.apache.shardingsphere.data.pipeline.core.metadata.model;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.sql.Types;
+import java.util.List;
+import java.util.Map;
+
+public final class PipelineTableMetaDataTest {
+    private PipelineTableMetaData pipelineTableMetaData;
+    private Map<String, PipelineColumnMetaData> columnMetaDataMap;
+
+    @Before
+    public void setUp() {
+        PipelineColumnMetaData pipelineColumnMetaData = new PipelineColumnMetaData(0, "test", Types.INTEGER, "INTEGER", true);
+        columnMetaDataMap.put("test_column", pipelineColumnMetaData);
+        pipelineTableMetaData = new PipelineTableMetaData(null, columnMetaDataMap);
+    }
+
+    @Test
+    public void assertGetColumnMetaDataGivenColumnIndex() {
+        List<String> pipelineTableColumns = pipelineTableMetaData.getColumnNames();
+        PipelineColumnMetaData actual = columnMetaDataMap.get(pipelineTableColumns.get(0));
+        assertThat(actual.getOrdinalPosition(), is(0));
+        assertThat(actual.getName(), is("test"));
+        assertThat(actual.getDataType(), is(Types.INTEGER));
+        assertTrue(actual.isPrimaryKey());
+    }
+
+    @Test
+    public void assertGetColumnMetaDataGivenColumnName() {
+        PipelineColumnMetaData actual = columnMetaDataMap.get("test_column");
+        assertThat(actual.getOrdinalPosition(), is(0));
+        assertThat(actual.getName(), is("test"));
+        assertThat(actual.getDataType(), is(Types.INTEGER));
+        assertTrue(actual.isPrimaryKey());
+    }
```

Comment:
1. It's better to invoke `pipelineTableMetaData.getColumnMetaData` instead of `columnMetaDataMap.get("test_column")`, since the logic is encapsulated in `pipelineTableMetaData` and we'll test it.

2. It could be better to add a new case: get column metadata by non-existence column name.

Reply:
1. Thank you for pointing it out, I was a bit skeptical in using `columnMetaDataMap` for accessing the `test_column`. I have made the change.
2. I have added the new case -
```java
assertThat(pipelineTableMetaData.getColumnMetaData("not_exist"), is(-1));
```

Reply:
@sandynz I get a failing test case here while checking for a non-existent column since `getColumnMetaData` returns `null` here, not `-1`.

Reply:
You could try `assertNull(pipelineTableMetaData.getColumnMetaData("not_exist"))`.